test/e2e/util.go

```diff
@@ -522,49 +593,50 @@ func RunRC(c *client.Client, name string, ns, image string, replicas int) error
 	last = 0
 	failCount = 10
 	current = 0
+	podLists.Reset()
 	for same < failCount && current < replicas {
-		current = 0
-		waiting := 0
-		pending := 0
-		unknown := 0
-		time.Sleep(10 * time.Second)
-
-		currentPods, err := listPods(c, ns, label, fields.Everything())
-		if err != nil {
-			return fmt.Errorf("Error listing pods: %v", err)
-		}
-		if len(currentPods.Items) != len(pods.Items) {
-			return fmt.Errorf("Number of reported pods changed: %d vs %d", len(currentPods.Items), len(pods.Items))
-		}
-		for _, p := range currentPods.Items {
-			if p.Status.Phase == api.PodRunning {
-				current++
-				if err := VerifyContainersAreNotFailed(p); err != nil {
-					return err
-				}
-			} else if p.Status.Phase == api.PodPending {
-				if p.Spec.Host == "" {
-					waiting++
-				} else {
-					pending++
+		time.Sleep(2 * time.Second)
+		for podLists.Len() > 0 && current < replicas {
+			item := podLists.Pop()
+			current = 0
+			waiting := 0
+			pending := 0
+			unknown := 0
+			currentPods := item.value.([]api.Pod)
+			for _, p := range currentPods {
+				if p.Status.Phase == api.PodRunning {
+					current++
+					if err := VerifyContainersAreNotFailed(p); err != nil {
+						return err
+					}
+				} else if p.Status.Phase == api.PodPending {
+					if p.Spec.Host == "" {
+						waiting++
+					} else {
+						pending++
+					}
+				} else if p.Status.Phase == api.PodUnknown {
+					unknown++
 				}
-			} else if p.Status.Phase == api.PodUnknown {
-				unknown++
 			}
+			Logf("Pod States: %d running, %d pending, %d waiting, %d unknown", current, pending, waiting, unknown)
```

Comment:
I must have misread something. Now I see only one place where you write to file... Sorry.